test/widgets/message_list_test.dart

```diff
@@ -85,17 +85,23 @@ void main() {
       tester.widget<StickyHeaderListView>(find.byType(StickyHeaderListView)).semanticChildCount;
 
     testWidgets('basic', (tester) async {
+      // TODO(#445): tweak threshold for fetching more messages, revert counts to 100
+      //   Message batch sizes here changed to 200 -- an unnatural
+      //   result in the app given `kMessageListFetchBatchSize`
+      //   is 100. This is necessary in this test as the new messages
+      //   loaded are shorter than the threshold for fetching more messages
+      //   causing an extra request to be made.
       await setupMessageListPage(tester, foundOldest: false,
-        messages: List.generate(200, (i) => eg.streamMessage(id: 950 + i, sender: eg.selfUser)));
-      check(itemCount(tester)).equals(203);
+        messages: List.generate(150, (i) => eg.streamMessage(id: 950 + i, sender: eg.selfUser)));
+      check(itemCount(tester)).equals(153);
```

Comment:
Here's a simpler solution:

```suggestion
        messages: List.generate(300, (i) => eg.streamMessage(id: 950 + i, sender: eg.selfUser)));
      check(itemCount(tester)).equals(303);
```

and then the line at the end of the test expects 403 items instead of 303.

This way the new batch when it fetches more can remain 100 messages like it is now. That also makes the apologetic comment unnecessary — the message list starts with 300 messages, which is more than 100, but it's perfectly normal for a message list to have 300 messages in it if the user has already scrolled up some.

I think the key to what's going on here isn't actually anything untoward in how we're deciding when to fetch more messages. Rather, it's this line of the test:
```dart
      await tester.pump(const Duration(seconds: 3)); // Fast-forward to end of fling.
```
where we don't let any app code run between when the user starts a fling scroll (one at full speed, 8000 px/sec), and three seconds later when the fling scroll should be complete. With 200 very compact example messages (short contents, and all from the same sender to the same recipient on the same day), it's likely we've scrolled past all of them by that point. In normal operation we'd have started the next fetch long before then, but with this fast-forward we don't get the chance.